legal-api/src/legal_api/services/filings/validations/incorporation_application.py

```diff
@@ -186,3 +192,40 @@ def validate_share_structure(incorporation_json) -> Error:
         return msg
 
     return None
+
+
+def validate_incorporation_effective_date(incorporation_json) -> Error:
+    """Return an error or warning message based on the effective date validation rules.
+
+    Rules:
+        - The effective date must be the correct format.
+        - The effective date must be a minimum of 2 minutes in the future.
+        - The effective date must be a maximum of 10 days in the future.
+    """
+    # Setup
+    msg = []
+    now = dt.utcnow()
+    now_plus_2_minutes = now + timedelta(minutes=2)
+    now_plus_10_days = now + timedelta(days=10)
+
+    try:
+        filing_effective_date = incorporation_json['filing']['header']['effectiveDate']
+    except KeyError:
+        return msg
+
+    try:
+        effective_date = dt.fromisoformat(filing_effective_date)
+    except ValueError:
+        msg.append({'error': babel('%s is an invalid ISO format for effective_date.') % filing_effective_date})
+        return msg
+
+    if effective_date < now_plus_2_minutes:
```

Comment:
The front end actually stores and makes the request with `effectiveDate` it in UTC('IsoString').
 Just from working with DateTimes earlier, we knew that since the backend was handling it in this format, it was simpler to avoid the time conversions. We just convert convert the DateTime when presenting it in the UI.

Reply:
awesome